homeassistant/components/proxmoxve/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config Flow for ProxmoxVE."""
+import logging
+
+import proxmoxer
+from requests.exceptions import ConnectTimeout, SSLError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.data_entry_flow import FlowResult
+
+from . import ProxmoxClient
+from .const import CONF_REALM, DEFAULT_PORT, DEFAULT_REALM, DEFAULT_VERIFY_SSL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ProxmoxVEConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """ProxmoxVE Config Flow class."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Init for ProxmoxVE config flow."""
+        super().__init__()
+
+        self._config = {}
+        self._proxmox_client = None
+
+    async def async_step_import(self, import_config=None):
+        """Import existing configuration."""
+        return await self.async_step_init(import_config, True)
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Manual user configuration."""
+        return await self.async_step_init(user_input, False)
+
+    async def async_step_init(self, info, is_import):
+        """Async step user for proxmoxve config flow."""
+        errors = {}
+
+        if info is not None:
+
+            host = info.get(CONF_HOST, "")
+            port = info.get(CONF_PORT, DEFAULT_PORT)
+            username = info.get(CONF_USERNAME, "")
+            password = info.get(CONF_PASSWORD, "")
+            realm = info.get(CONF_REALM, DEFAULT_REALM)
+            verify_ssl = info.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL)
+
+            if await self._async_endpoint_exists(f"{host}/{port}"):
+                _LOGGER.info(
+                    "Device %s already configured, you can remove it from configuration.yaml if you still have it",
+                    host,
+                )
+                return self.async_abort(reason="already_configured")
+
+            if port > 65535 or port <= 0:
+                errors[CONF_PORT] = "invalid_port"
+
+            if not errors:
+
+                try:
+                    self._proxmox_client = ProxmoxClient(
+                        host,
+                        port=port,
+                        user=username,
+                        realm=realm,
+                        password=password,
+                        verify_ssl=verify_ssl,
+                    )
+
+                    await self.hass.async_add_executor_job(
+                        self._proxmox_client.build_client
+                    )
+
+                except proxmoxer.backends.https.AuthenticationError:
+                    errors[CONF_USERNAME] = "auth_error"
```

Comment:
This doesn't have to be username specific, maybe use the base instead

```suggestion
                    errors["base"] = "auth_error"
```